libevmasm/EVMAssemblyStack.h

```diff
@@ -26,15 +26,23 @@
 
 #include <map>
 #include <string>
+#include <utility>
 
 namespace solidity::evmasm
 {
 
 class EVMAssemblyStack: public AbstractAssemblyStack
 {
 public:
-	explicit EVMAssemblyStack(langutil::EVMVersion _evmVersion, std::optional<uint8_t> _eofVersion):
-		m_evmVersion(_evmVersion), m_eofVersion(_eofVersion) {}
+	EVMAssemblyStack(
+		langutil::EVMVersion _evmVersion,
+		std::optional<uint8_t> _eofVersion,
+		frontend::OptimiserSettings _optimiserSettings
```

Comment:
Wouldn't it be more appropriate to do the translation on the other side and already pass in ``Assembly::OptimiserSettings`` here? It's an Assembly Stack after all...

Reply:
Fully agree, fixed